client/packages/system/src/Patient/api/hooks/document/usePatients.ts

```diff
@@ -3,10 +3,20 @@ import { usePatientApi } from '../utils/usePatientApi';
 
 export const usePatients = () => {
   const api = usePatientApi();
-  const { queryParams } = useUrlQueryParams({
+  const { queryParams, urlQuery } = useUrlQueryParams({
     filterKey: 'lastName',
     initialSort: { key: 'code', dir: 'asc' },
   });
+  if (urlQuery['firstName'] || `${urlQuery['identifier']}`) {
+    const filterBy = queryParams.filterBy ?? {};
+    if (urlQuery['firstName']) {
+      filterBy['firstName'] = { like: `${urlQuery['firstName']}` };
+    }
+    if (urlQuery['identifier']) {
+      filterBy['identifier'] = { like: `${urlQuery['identifier']}` };
+    }
+    queryParams.filterBy = filterBy;
+  }
```

Comment:
is there a nicer way to do this?

Reply:
likely would have to implement multiple field `filterKey` in `useUrlPerParams` - there's nothing at the moment to do this in a nicer way. Perhaps if we specified `filterKeys` as an array of keys instead of a single string